### ISSUE TITLE
Update documentation on how to enable the Data Plane API in 3.1 and above

### ISSUE THREAD

**Issue by WIStudent** [haproxytech/haproxy-docker-alpine#14]
[HAProxy 3.1 deprecated the `program` section](https://www.haproxy.com/blog/announcing-haproxy-3-1#deprecated-features), so to enable the Data Plane API as [described in the readme](https://github.com/haproxytech/haproxy-docker-alpine?tab=readme-ov-file#enable-data-plane-api) you need to set the `expose-deprecated-directives` global option.

```
global
   expose-deprecated-directives

program api
   command /usr/bin/dataplaneapi --host 0.0.0.0 --port 5555 --haproxy-bin /usr/sbin/haproxy --config-file /usr/local/etc/haproxy/haproxy.cfg --reload-cmd "kill -SIGUSR2 1" --restart-cmd "kill -SIGUSR2 1" --reload-delay 5 --userlist haproxy-dataplaneapi
   no option start-on-reload
```

The deprecation announcement also states that the `program` section will be removed in HAProxy 3.3. Please update the readme with instructions on how to use the Data Plane API in this image without the `program` section.

**Comment by dkorunic** [comment]
This is correct, using `program` was pretty much never the best choice compared to for instance s6 overlay. However we didn't want to add s6 overlay by default due to many people using our images for their own projects, and enforcing a single solution is definitely not an ideal approach. I'll detail how to approach this and install s6 overlay, together with examples of configs, similar to how we do it in [Ingress Controller](https://github.com/haproxytech/kubernetes-ingress/tree/master/fs/etc/s6-overlay) project.

**Comment by dkorunic** [comment]
We plan to offer s6-enabled api variant (having HAProxy and Dataplaneapi as s6 services) of the image really soon, within a week @WIStudent

**Comment by pbnsh** [comment]
@dkorunic Are "s6-enabled api variant" images available on docker hub? Docs still mention `program`.

**Comment by dkorunic** [comment]
@pbnsh It's work in progress as we speak, coming very soon.
